pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferenceManagerWorkerService.scala

```diff
@@ -38,48 +50,94 @@ class InferenceManagerWorkerService[Destination](
           case (_, image) => image
         }
         .via(imageDownloader.download)
-        .via(createRequest)
+        .via(createRequests)
         .via(requestPool.asContextFlow)
         .via(unmarshalResponse)
-        .via(augmentInput)
+        .via(collectAndAugment)
         .via(sendAugmented)
         .asSource
         .map { case (_, msg) => msg }
     )
 
-  private def createRequest[Ctx] =
-    FlowWithContext[DownloadedImage, Ctx].map { image =>
-      (inferrerAdapter.createRequest(image), image)
+  private def createRequests[Ctx] =
+    FlowWithContext[DownloadedImage, Ctx].mapConcat { image =>
```

Comment:
`FlowWithContext` ... new to me, very handy.

Reply:
As with lots of things in akka-streams it has some non-obvious restrictions in order to keep it theoretically pure, but on the whole it definitely is handy